tests/integration/helpers/helpers.go

```diff
@@ -0,0 +1,221 @@
+func ModifyJSONSchemaMetricsPresent(schema *simplejson.Json, eventType string) error {
+	mainProperties := schema.Get("properties")
+
+	elementOfMetrics := map[string]schemaElement{
+		"metrics": {"minItems", 1},
+	}
+	err := addNewElementsToJSONSchema(mainProperties, elementOfMetrics)
+	if err != nil {
+		return err
+	}
+
+	metrics, ok := mainProperties.CheckGet("metrics")
+	if !ok {
+		return fmt.Errorf("Cannot find metrics element")
+	}
+	metricsItems, ok := metrics.CheckGet("items")
+	if !ok {
+		return fmt.Errorf("Cannot find metrics items element")
+	}
+	metricsItemsProperties := metricsItems.Get("properties")
+	elementsOfMetricsProperties := map[string]schemaElement{
+		"event_type": {"pattern", fmt.Sprintf("^%s$", eventType)},
+	}
+	err = addNewElementsToJSONSchema(metricsItemsProperties, elementsOfMetricsProperties)
+	if err != nil {
+		return err
+	}
+
+	items, err := metricsItems.Map()
+	if err != nil {
+		return fmt.Errorf("Not expected metrics structure for 'item' element, got error: %v", err)
+	}
+	tmp := []interface{}{items}
+	metrics.Set("items", tmp)
+
+	// TODO: iterate through list of items and modify each itemSet by adding event_type pattern
+	// itemArray, err := metrics.Get("items").Array()
+	// for _, itemSet := range itemArray {
+	// 	jsonSet := simplejson.Json{itemSet} // fails
+	// }
+
+	return nil
+}
+
+// ModifyJSONSchemaNoMetrics modifies JSON schema by adding required elements
+// assuring that no metrics data exists in the integration output
+func ModifyJSONSchemaNoMetrics(schema *simplejson.Json) error {
+	return addNewElementsToJSONSchema(
+		schema.Get("properties"),
+		map[string]schemaElement{
+			"metrics": {"maxItems", 0},
+		})
+}
+
+// ModifyJSONSchemaNoInventory modifies JSON schema by adding required elements
+// assuring that no inventory data exists in the integration output
+func ModifyJSONSchemaNoInventory(schema *simplejson.Json) error {
+	return addNewElementsToJSONSchema(
+		schema.Get("properties"),
+		map[string]schemaElement{
+			"inventory": {"maxProperties", 0},
+		})
+}
+
+// ModifyJSONSchemaInventoryPresent modifies JSON schema by adding required elements for inventory JSON schema
+func ModifyJSONSchemaInventoryPresent(schema *simplejson.Json) error {
+	mainProperties := schema.Get("properties")
+
+	elementOfMainInventory := map[string]schemaElement{
+		"inventory": {"minProperties", 1},
+	}
+	err := addNewElementsToJSONSchema(mainProperties, elementOfMainInventory)
+	if err != nil {
+		return err
+	}
+
+	// `required` section for `inventory` should be empty, as none of the inventory data is obligatory
+	mainProperties.Get("inventory").Set("required", make([]interface{}, 0))
+	return nil
+}
+
+// ExecInContainer executes the given command inside the specified container. It returns three values:
+// 1st - Standard Output
+// 2nd - Standard Error
+// 3rd - Runtime error, if any
+func ExecInContainer(container string, command []string, envVars ...string) (string, string, error) {
```

Comment:
why do we need stdout & stderr, would not be simpler returning only error from `cmd.Run`. From the API design point of view is simpler and it helps to understand if there was an error while executing the container, WDYT?

Reply:
stdout is obviously necessary, I was referring including stderr in the error

Reply:
As a generic command, I returned everything to allow more insights in case of error (e.g. showing the standard error).

We could remove the standard error but at least the standard output is needed to parse the JSON output of the integration.

Reply:
yep, indeed we need the stdout. Let's keep it like this for now. 👍